### PR TITLE
Refactor image build check 

### PR DESCRIPTION
**Pull request by rokroskar**
Move the logic regarding image selection out of the spawner and into the notebook service. The spawner should just launch whichever image is given to it.

### CODE REVIEW

**Comment by rokroskar** [comment]
This seems to work, tested on staging. Would be good to have a second pair of eyes.

File: helm-chart/renku-notebooks/values.yaml [hunk 1]
```diff
@@ -89,8 +96,6 @@ jupyterhub:
     extraEnv:
       DEBUG: 1
       JUPYTERHUB_SPAWNER_CLASS: spawners.RenkuKubeSpawner
-      ## timeout for waiting on an image build
-      # GITLAB_IMAGE_BUILD_TIMEOUT: 600
```
Comment (leafty) [inline helm-chart/renku-notebooks/values.yaml]: Need to move it instead, no?

**Comment by leafty** [comment]
I see. I guess it might be good to rethink the PR then. Let's see how we can have public image selection together with repo-based images.

**Comment by rokroskar** [comment]
@leafty I am happy to remove that part for now as it doesn't really address the point of this PR in any case. We may want to look at the JupyterHub spawner `options_form` feature which allows you to do something along these lines and control which images are allowed.

File: helm-chart/renku-notebooks/templates/deployment.yaml [hunk 2]
```diff
@@ -48,6 +48,14 @@ spec:
               value: {{ .Values.jupyterhub.hub.services.notebooks.oauth_client_id }}
             - name: GITLAB_REGISTRY_SECRET
               value: {{ template "renku.fullname" . }}-registry
+            - name: GITLAB_URL
+            {{ if .Values.gitlab_url }}
```
Comment (jirikuncar) [inline helm-chart/renku-notebooks/templates/deployment.yaml]: `.Values.gitlab.url` (see https://github.com/SwissDataScienceCenter/renku-notebooks/pull/56/files#diff-4a2cd40c069f23f7cd71ab6a74028709R10)?

**Comment by rokroskar** [comment]
@leafty can you please review again?

File: helm-chart/renku-notebooks/templates/deployment.yaml [hunk 3]
```diff
@@ -28,26 +28,36 @@ spec:
...
           env:
+            - name: DEFAULT_NOTEBOOK_IMAGE
+              value: "{{ .Values.jupyterhub.singleuser.image.name }}:{{ .Values.jupyterhub.singleuser.image.tag }}"
             {{ if eq .Values.debug true }}
             - name: FLASK_DEBUG
               value: "1"
             {{ end }}
             - name: JUPYTERHUB_API_TOKEN
-              value: {{ .Values.jupyterhub_api_token }}
+              value: {{ .Values.jupyterhub.hub.services.notebooks.api_token }}
```
Comment (leafty) [inline helm-chart/renku-notebooks/templates/deployment.yaml]: :+1:

File: singleuser/Dockerfile [hunk 4]
```diff
@@ -1,12 +1,12 @@
 # Build as renku/singleuser
 # Run with the DockerSpawner in JupyterHub
 
-ARG JUPYTERHUB_VERSION=0.9
+ARG JUPYTERHUB_VERSION=0.9.2
```
Comment (leafty) [inline singleuser/Dockerfile]: they made `0.9.1` disappear, will this one last?

Comment (rokroskar) [reply singleuser/Dockerfile]: it's there: https://github.com/jupyterhub/jupyterhub/releases/tag/0.9.1

**Pull request merged**